tools/fastdds/xml_ci/validate.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Sub-Command Validate implementation.
+
+This sub-command validates the Fast DDS XML configuration files using an
+XSD schema.
+"""
+
+import os
+
+try:  # Try catch for new python dependency
+    from xml import etree
+except ImportError:
+    print(
+        'xml module not found. Try to install running "pip install xml"')
+    exit(1)
+
+try:  # Try catch for new python dependency
+    from xmlschema import XMLSchema, XMLSchemaValidationError
+except ImportError:
+    print(
+        'xmlschema module not found. Try to install running "pip install xmlschema"')
+    exit(1)
+
+
+class Validate:
+    """Fast DDS XML configuration files validator."""
+
+    def __init__(self, xsd_file):
+        """Initialize Validate class."""
+        self.xsd_file = xsd_file
+        self.__visited_dirs = []
+
+    @property
+    def xsd_file(self):
+        """XSD schema getter."""
+        return self.__xsd_file
+
+    @xsd_file.setter
+    def xsd_file(self, xsd_file):
+        """XSD schema setter."""
+        if not os.path.isfile(xsd_file):
+            print(f'The XSD schema does not exist: {xsd_file}')
+            exit(1)
+        if isinstance(xsd_file, XMLSchema):
+            self.__xsd_file = xsd_file
+        else:
+            try:
+                self.__xsd_file = XMLSchema(xsd_file)
+            except etree.ElementTree.ParseError as e:
+                print(f'cannot load XSD file: {xsd_file}')
+                print(e)
+                exit(1)
+
+    def run(self, validate_list):
+        """
+        Run Fast DDS XML configuration files validation.
+
+        :param validate_list: List with the directories and files to validate.
+        :return: False if failed parsing, otherwise True.
+        """
+        valid = True
+        self.__visited_dirs.clear()
+        for element in validate_list:
+            element = os.path.abspath(element)
+            if os.path.isdir(element):
+                valid = valid and self.__validate_xml_from_dir(element)
+            elif os.path.isfile(element):
+                if self.__is_xml(element):
+                    valid = valid and self.__validate_xml_file(element)
+                else:
+                    print(f'The file is not an XML file: {element}')
+                    valid = False
+        return valid
```

Comment:
I have encountered that, if I enter an incorrect xml file name for the input argument, the tool doesn't print any output. 
It should say that it is not an xml file at least.